Fixed loading translation values for User Profile attributes

agagancarczyk opened the pull request:
Closes https://github.com/keycloak/keycloak/issues/27837

agagancarczyk committed:
fixed displaying default translations

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

agagancarczyk committed:
fixed the issue translations

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

agagancarczyk committed:
improvement

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

agagancarczyk commented on js/apps/admin-ui/src/realm-settings/user-profile/attribute/AddTranslationsDialog.tsx:
```diff
@@ -105,15 +108,60 @@ export const AddTranslationsDialog = ({
     );
   }, [combinedLocales, filter, whoAmI]);
 
+  useEffect(() => {
```
sounds good @jonkoops

ssilvert reviewed (changes requested):
I tried the code out and it's not working.  When I click on the button to bring up the translations dialog I get a NullPointerException on the server.  I am using the latest nightly build of the server.
`Caused by: java.lang.NullPointerException: Cannot invoke "org.keycloak.models.UserSessionModel.getUser()" because "userSession" is null`

The request that is failing is `http://localhost:8180/admin/realms/test/localization/en`.
500 Internal Server Error

@agagancarczyk Maybe try this with the latest nightly and see if you see this too?

jonkoops commented:
@ssilvert I've also been seeing that error locally in other places, it is related how things are proxied during development. It should work if you do a full production build and test against that. Unfortunately, this also makes it impossible to run the Cypress tests on some suites against the development server.

I think we should unify the dev server and the Keycloak server, so that the Keycloak server always is responsible for serving the front-end, even during development. I've started #27311 as a thought experiment on how to accomplish this.

agagancarczyk committed:
added logic for scenario 2

agagancarczyk committed:
added logic for scenario 2

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

ssilvert commented:
@agagancarczyk @jonkoops OK, I was able to run the code by doing a full build and scenario 1 is fixed.

I am still seeing scenario 2 from the issue.  Agi, note that I updated the steps for scenario 2 as I was unclear in my original post.

I also added scenario 3, which is another related problem.  See the issue.

ssilvert commented:
> I think we should unify the dev server and the Keycloak server, so that the Keycloak server always is responsible for serving the front-end, even during development. I've started #27311 as a thought experiment on how to accomplish this.

@jonkoops That would be amazing if you can pull it off.  It would also solve some other problems I've seen such as changes to message bundles that are not being reflected in dev mode.

jonkoops commented:
I'll try to put some time into it when we're done with the PatternFly migration :+1:

agagancarczyk commented:
> @agagancarczyk @jonkoops OK, I was able to run the code by doing a full build and scenario 1 is fixed.
> 
> I am still seeing scenario 2 from the issue. Agi, note that I updated the steps for scenario 2 as I was unclear in my original post.
> 
> I also added scenario 3, which is another related problem. See the issue.

@ssilvert scenario 2 has also been fixed. Can you try again? I pushed a fix yesterday. Maybe you are expecting something else? I'll take a look at scenario 3.

agagancarczyk committed:
improvement

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

agagancarczyk committed:
added logic for scenario 3

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

agagancarczyk committed:
refactor

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

agagancarczyk committed:
refactor

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

agagancarczyk committed:
refactor

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

agagancarczyk commented:
Closing this PR as it's old in favour of https://github.com/keycloak/keycloak/pull/28899
